File: python/two_sum.py

def two_sum(nums:list, target: int) -> int:
    num_map: dict = {}
    for i, num in enumerate(nums):
        print(enumerate(nums))
        compliment = target - num
        if compliment in num_map:
            return [num_map[compliment], i]
        num_map[num] = i
        
        
numbers = [1, 2, 3, 99, 4, 101, 0, 5, 6, 9, 3]
print(two_sum(numbers, 9))

# Time Complexity = 0(n)
# Space complexity 0(n)

File: javascript/twoSum.js
function twoSum(nums, target) {
  const numsMap = new Map();
  for (let i = 0; i < nums.length; i++) {
    const compliment = target - nums[i];
    if (numsMap.has(compliment)) {
      return [numsMap.get(compliment), i];
    }
    numsMap.set(nums[i], i);
  }
  return [];
}

numbers = [1, 2, 3, 99, 4, 101, 0, 5, 6, 9, 3];
console.log(twoSum(numbers, 9));
